frontend/src/components/dashboard/Analyzer.tsx

```diff
@@ -166,111 +219,166 @@ const Analyzer = ({ session }: { session: ExistingSessionExtended }) => {
-              taskType={task.type}
-              xAxis={xAxis}
-              setXAxis={updateXAxis}
-              yAxis={yAxis}
-              setYAxis={updateYAxis}
-              categorizedDataPoints={categorizedDataPoints}
-              manualGroups={manualGroups}
-              splittingEnabled={!isAutomaticGrouping}
-              splits={splits}
-              setSplits={setSplits}
-            />
-          </Col>
-          <Col xs={12}>
-            {task && (
-              <CodeComparison
-                classId={session.klass.id}
-                sessionId={session.id}
-                taskId={task.id}
-                subTaskId={selectedSubTaskId}
+                {isAutomaticGrouping && (
+                  <Input
+                    label={messages.numberOfGroups}
+                    type="number"
+                    value={numberOfGroups}
+                    onChange={(e) =>
+                      setNumberOfGroups(parseInt(e.target.value))
+                    }
+                  />
+                )}
+
+                {!isGroupingAvailable && (
+                  <FormattedMessage
+                    id="Analyzer.noGroupingAvailable"
+                    defaultMessage="Computing groups, please be patient."
+                  />
+                )}
+              </Parameters>
+            </Col>
+            <Col xs={12} lg={9}>
+              <Analysis
                 taskType={task.type}
-                groupAssignments={groupAssignments}
-                groups={groups}
+                xAxis={xAxis}
+                setXAxis={updateXAxis}
+                yAxis={yAxis}
+                setYAxis={updateYAxis}
+                categorizedDataPoints={categorizedDataPoints}
+                manualGroups={manualGroups}
+                splittingEnabled={!isAutomaticGrouping}
+                splits={splits}
+                setSplits={setSplits}
+                onSelectSolution={onSelectSolution}
               />
-            )}
-          </Col>
-        </Row>
-      )}
-    </MultiSwrContent>
+            </Col>
+            <Col xs={12}>
+              {task && (
+                <CodeComparison
+                  classId={session.klass.id}
+                  sessionId={session.id}
+                  taskId={task.id}
+                  subTaskId={selectedSubTaskId}
+                  taskType={task.type}
+                  groupAssignments={groupAssignments}
+                  groups={groups}
+                  selectedLeftGroup={selectedLeftGroup}
+                  setSelectedLeftGroup={setSelectedLeftGroup}
+                  selectedRightGroup={selectedRightGroup}
+                  setSelectedRightGroup={setSelectedRightGroup}
+                  selectedLeftSolution={selectedLeftSolution}
+                  setSelectedLeftSolution={setSelectedLeftSolution}
+                  selectedRightSolution={selectedRightSolution}
+                  setSelectedRightSolution={setSelectedRightSolution}
```

Comment:
could we define an interface and pass that around ? e.g.
```
interface SelectedSolution {
  group: string;
  setGroup(s: string): void;
  solution: string;
  setSolution(s: string): void;
}
```
and then just pass `leftSelected` and `rightSelected` ?
This seems very repetitive and hence error-prone (we've already caught a few issues where left and right were swapped).

Reply:
Is improved in a later PR :)